File: src/main/java/red/tetracube/iot/smartigloohub/data/repositories/TrustedBrokerClientRepository.java
package red.tetracube.iot.smartigloohub.data.repositories;

import io.smallrye.mutiny.Uni;
import org.hibernate.reactive.mutiny.Mutiny;
import red.tetracube.iot.smartigloohub.data.entities.TrustedBrokerClient;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;

@ApplicationScoped
public class TrustedBrokerClientRepository {

    @Inject
    Mutiny.SessionFactory rxSessionFactory;

    public Uni<TrustedBrokerClient> getBrokerClientByClientIdAndUsername(String clientId, String username) {
        return rxSessionFactory.withSession(session ->
                session.createQuery("""
                                            from TrustedBrokerClient t
                                            where t.clientId = :clientId and t.username = :username
                                        """,
                                TrustedBrokerClient.class
                        )
                        .setParameter("clientId", clientId)
                        .setParameter("username", username)
                        .setMaxResults(1)
                        .getSingleResultOrNull()
        );
    }

}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/data/repositories/TrustedBrokerClientRepository.kt
package red.tetracube.iot.smartigloohub.data.repositories

import io.smallrye.mutiny.Uni
import org.hibernate.reactive.mutiny.Mutiny
import red.tetracube.iot.smartigloohub.data.entities.TrustedBrokerClient
import javax.enterprise.context.ApplicationScoped

@ApplicationScoped
class TrustedBrokerClientRepository(
    private val rxSessionFactory: Mutiny.SessionFactory
) {

    fun getBrokerClientByClientIdAndUsername(clientId: String, username: String): Uni<TrustedBrokerClient?> {
        return rxSessionFactory.withSession { sesssion ->
            sesssion.createQuery(
                """
                from TrustedBrokerClient t
                where t.clientId = :clientId and t.username = :username
            """.trimIndent(),
                TrustedBrokerClient::class.java
            )
                .setParameter("clientId", clientId)
                .setParameter("username", username)
                .setMaxResults(1)
                .singleResultOrNull
        }
    }
}